qcore/geo.py

```diff
@@ -38,16 +41,18 @@ def ll2gp_multi(coords, mlon, mlat, rot, nx, ny, hh, \
 
     # run binary, get output
     # output is displacement (x, y) from center, in kilometres
-    cmd = [ll2xy_bin, 'mlat=%s' % (mlat), 'mlon=%s' % (mlon), \
-              'geoproj=1', 'center_origin=1', 'h=%s' % (hh), \
-              'xazim=%s' % (xazim), 'xlen=%s' % (xlen), 'ylen=%s' % (ylen)]
+    cmd = [ll2xy_bin, 'mlat=%s' % (mlat), 'mlon=%s' % (mlon),
+           'geoproj=1', 'center_origin=1', 'h=%s' % (hh),
+           'xazim=%s' % (xazim), 'xlen=%s' % (xlen), 'ylen=%s' % (ylen)]
     if verbose:
         print(' '.join(cmd))
-    p_conv = Popen(cmd,
-            stdin = PIPE, stdout = PIPE)
-    stdout = p_conv.communicate( \
-            '\n'.join(['%s %s' % tuple(c) for c in coords]))[0]
-    xy = [map(float, line.split()) for line in stdout.rstrip().split('\n')]
+
+    # Has to be a byte string
+    p_conv = Popen(cmd, stdin=PIPE, stdout=PIPE)
+    stdout = p_conv.communicate(
+        '\n'.join(['%s %s' % tuple(c) for c in coords]).encode())[0]
+    xy = [list(map(float, line.split()))
+          for line in stdout.rstrip().split(b'\n')]
```

Comment:
i guess considering the final output is actually a float, converted then it's ok to not run decode as long as this part works fine.